commit 0e311d33bfbef86da130674e8528cc23e6acfe16
Author: Suravee Suthikulpanit <suravee.suthikulpanit@amd.com>
Date:   Thu May 19 05:27:05 2022 -0500

    KVM: SVM: Introduce hybrid-AVIC mode
    
    Currently, AVIC is inhibited when booting a VM w/ x2APIC support.
    because AVIC cannot virtualize x2APIC MSR register accesses.
    However, the AVIC doorbell can be used to accelerate interrupt
    injection into a running vCPU, while all guest accesses to x2APIC MSRs
    will be intercepted and emulated by KVM.
    
    With hybrid-AVIC support, the APICV_INHIBIT_REASON_X2APIC is
    no longer enforced.
    
    Suggested-by: Maxim Levitsky <mlevitsk@redhat.com>
    Reviewed-by: Maxim Levitsky <mlevitsk@redhat.com>
    Signed-off-by: Suravee Suthikulpanit <suravee.suthikulpanit@amd.com>
    Message-Id: <20220519102709.24125-14-suravee.suthikulpanit@amd.com>
    Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>

diff --git a/arch/x86/include/asm/kvm_host.h b/arch/x86/include/asm/kvm_host.h
index 88a3026ee163..de5a149d0971 100644
--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -1129,11 +1129,6 @@ enum kvm_apicv_inhibit {
 	 */
 	APICV_INHIBIT_REASON_PIT_REINJ,
 
-	/*
-	 * AVIC is inhibited because the guest has x2apic in its CPUID.
-	 */
-	APICV_INHIBIT_REASON_X2APIC,
-
 	/*
 	 * AVIC is disabled because SEV doesn't support it.
 	 */